pulser-simulation/pulser_simulation/hamiltonian.py

```diff
@@ -106,95 +105,38 @@ def config(self) -> NoiseModel:
-                )
-            # Probability of phase (Z) flip:
-            # First order in prob
-            prob = config.dephasing_prob / 2
-            n = self._size
-            if prob > 0.1 and n > 1:
-                warnings.warn(
-                    "The dephasing model is a first-order approximation in the"
-                    f" dephasing probability. p = {2*prob} is too large for "
-                    "realistic results.",
-                    stacklevel=2,
+                    f"Cannot include {noise_type} "
+                    + "noise in digital- or all-basis."
                 )
-            k = np.sqrt(prob * (1 - prob) ** (n - 1))
 
-            self._collapse_ops += [
-                np.sqrt((1 - prob) ** n)
-                * qutip.tensor([self.op_matrix["I"] for _ in range(n)])
-            ]
-            kraus_ops.append(k * qutip.sigmaz())
+        local_collapse_ops = []
+        if "dephasing" in config.noise_types:
+            basis_check("dephasing")
+            coeff = np.sqrt(config.dephasing_rate / 2)
+            local_collapse_ops.append(coeff * qutip.sigmaz())
 
         if "depolarizing" in config.noise_types:
-            if self.basis_name == "digital" or self.basis_name == "all":
-                # Go back to previous config
-                raise NotImplementedError(
-                    "Cannot include depolarizing "
-                    + "noise in digital- or all-basis."
-                )
-            # Probability of error occurrence
-
-            prob = config.depolarizing_prob / 4
-            n = self._size
-            if prob > 0.1 and n > 1:
-                warnings.warn(
-                    "The depolarizing model is a first-order approximation"
-                    f" in the depolarizing probability. p = {4*prob}"
-                    " is too large for realistic results.",
-                    stacklevel=2,
-                )
-
-            k = np.sqrt((prob) * (1 - 3 * prob) ** (n - 1))
-            self._collapse_ops += [
-                np.sqrt((1 - 3 * prob) ** n)
-                * qutip.tensor([self.op_matrix["I"] for _ in range(n)])
-            ]
-            kraus_ops.append(k * qutip.sigmax())
-            kraus_ops.append(k * qutip.sigmay())
-            kraus_ops.append(k * qutip.sigmaz())
+            basis_check("dephasing")
+            coeff = np.sqrt(config.depolarizing_rate / 4)
+            local_collapse_ops.append(coeff * qutip.sigmax())
+            local_collapse_ops.append(coeff * qutip.sigmay())
+            local_collapse_ops.append(coeff * qutip.sigmaz())
 
         if "eff_noise" in config.noise_types:
-            if self.basis_name == "digital" or self.basis_name == "all":
-                # Go back to previous config
-                raise NotImplementedError(
-                    "Cannot include general "
-                    + "noise in digital- or all-basis."
+            basis_check("general")
```

Comment:
Yeah that's what we were doing... It would make more sense to display it at "effective" instead.

Reply:
I don't know about "general", "arbitrary" or "custom" would make more sense to me. In any case, we are not doing the renaming here so we could defer the decision to a future PR. I'm just afraid that if you change to "general" here and then we decide to go with something else, this might be forgotten.

Reply:
So we were already going for "General" in several places, but I find it does not make much sense... This is why I decided to go with "effective" instead